### PR TITLE
Update and complete Gdal warp algorithm description

### PR DESCRIPTION
**Pull request by DelazJ**
documentation source: https://www.gdal.org/gdalwarp.html

- [x] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 1]
```diff
@@ -142,11 +151,36 @@ Parameters
 
   Default: *0*
 
+``Georeferenced extents of output file to be created (xmin, xmax, ymin, ymax)`` [extent]
+  Optional
+
+  Sets georeferenced extents of output file to be created (in ``target CRS`` by default or in ``CRS of the target raster extent`` if specified).
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  Sets georeferenced extent of the output file to be created (``target CRS`` by default. ``CRS of the target raster extent``, if specified).
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 2]
```diff
@@ -142,11 +151,36 @@ Parameters
...
+
+``CRS of the target raster extent`` [crs]
+  Optional
+
+  Specifies the CRS in which to interpret the coordinates given for the extents
+  of output file. This must not be confused with the target CRS of the output
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  of the output file. This must not be confused with the target CRS of the output
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 3]
```diff
@@ -142,11 +151,36 @@ Parameters
...
+  dataset. It is instead a convenience e.g. when knowing the output coordinates
+  in a geodetic long/lat SRS, but still wanting a result in a projected coordinate system.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  in a geodetic long/lat CRS, but wanting a result in a projected coordinate system.
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 4]
```diff
@@ -142,11 +151,36 @@ Parameters
...
+
+``Use multithreaded warping implementation`` [boolean]
+  Two threads will be used to process chunks of image and perform input/output
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  Two threads will be used to process chunks of the image and perform input/output
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 5]
```diff
@@ -142,11 +151,36 @@ Parameters
...
+  operation simultaneously. Note that computation is not multithreaded itself.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  operations simultaneously. Note that computation is not multithreaded itself.
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 6]
```diff
@@ -142,11 +151,36 @@ Parameters
...
+
+  Default: *False*
+
+``Additional command line parameters`` [string]
+  Optional
+
+  Allows to add more advanced reprojection parameters.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  Allows adding more advanced reprojection parameters.
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst (hunk 1 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]: I'm not convinced by `(``target CRS`` by default. ``CRS of the target raster extent``, if specified)` . It looks like if the extent was the `target crs` though it's expressed `IN the target crs`. Then the full stop and sentence that follows look weird to me

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst [hunk 7]
```diff
@@ -142,11 +151,36 @@ Parameters
...
 
+``Georeferenced extents of output file to be created (xmin, xmax, ymin, ymax)`` [extent]
+  Optional
+
+  Sets georeferenced extents of output file to be created (in ``target CRS`` by default or in ``CRS of the target raster extent`` if specified).
+
+``CRS of the target raster extent`` [crs]
+  Optional
+
+  Specifies the CRS in which to interpret the coordinates given for the extents
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]:
```suggestion
  Specifies the CRS in which to interpret the coordinates given for the extent
```

File: source/docs/user_manual/processing_algs/gdal/rasterprojections.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rasterprojections.rst]: My mistake - sorry about that!  I have tried to update.

**Pull request merged**